### ISSUE TITLE
API: /api/v1/config/fingerbank_setting/upstream doesn't handle error with proxies

### ISSUE THREAD

**Issue by nqb** [inverse-inc/packetfence#6277]
**Describe the bug**
If you configure PacketFence to use Fingerbank behind a proxy and your proxy doesn't work, you will get following error message from API:

```json
{"errors":[{"field":"api_key","message":"Invalid API key provided"}],"message":"Unable to validate","status":422}
```

which is not true in this case.

**To Reproduce**
1. Configure a bad proxy in Fingerbank settings
2. Try to verify your key

=> You got error mentioned above.

**Expected behavior**
API should return an error related to proxy communication, not to API key.

**Additional context**
/api/v1/configurator/config/fingerbank_setting/upstream should be taken into account too.

**Comment by jrouzierinverse** [comment]
Can you retry I am supplying a more detailed error now.

**Comment by nqb** [comment]
Hello @jrouzierinverse,

Yes, it works. I got this error after I configured a bad proxy:

```
config/fingerbank_setting/upstream: api_key: Error while testing API key *REDACTED*. Error was: 500 error while CONNECT thru proxy: 500 Can't connect to 172.17.17.10:8888
```

**Comment by nqb** [comment]
@jrouzierinverse would it be possible to remove API key from error message ?

**Comment by jrouzierinverse** [comment]
Yes however this would require us create a new Fingerbank package.

**Comment by jrouzierinverse** [comment]
Fixed with fingerbank/perl-client@a63fb4e7e39e6a916922310400e5a41549ee74d1

**Comment by nqb** [comment]
See #6285 for next part.
